src/tactic/hint.lean

```diff
@@ -0,0 +1,81 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import tactic.tidy
+
+namespace tactic
+
+namespace hint
+
+/-- An attribute marking a `tactic unit` or `tactic string` which should be used by the `hint` tactic. -/
+meta def hint_attribute : user_attribute := {
+  name := `hint,
+  descr := "A tactic that should be tried by `hint`."
+}
+
+run_cmd attribute.register ``hint_attribute
+
+open lean lean.parser interactive
+
+private meta def add_tactic_hint (n : name) (t : expr) : tactic unit :=
+do
+  add_decl $ declaration.defn n [] `(tactic string) t reducibility_hints.opaque ff,
+  hint_attribute.set n () tt,
+  nolint_attr.set n () tt
```

Comment:
If the names become `tac._hint` then this line can be deleted, the linter will ignore them.

Reply:
Works nicely!